md/bdm-in-rest-api.md

```diff
@@ -216,6 +216,108 @@ class CarManagement implements RestApiController {
+At the end, in your rest API endpoint, you interact with the mapper.
+
+Here is an implementation example for the object Car which has four lazy attributes of type Wheel:  
+
+The serializer takes a list of Car in input, and build a Json object for each car. The wheels are replaced with links to an other Rest API extension with the car ID and the wheel number in parameter. Calling this API will return the wheel. This is a classic lazy behavior.
+```groovy
+/***********************
+ ***** SERIALIZER ******
+ ***********************/
+
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.databind.JsonSerializer
+import com.fasterxml.jackson.databind.SerializerProvider
+
+class CarSerializer extends JsonSerializer<List<Car>>{
+
+	@Override
+	public void serialize(List<Car> cars, JsonGenerator jgen, SerializerProvider provider)throws IOException, JsonProcessingException {
+		jgen.writeStartArray()
+		
+		cars.forEach({serializeCar(it,jgen)})
+		
+		jgen.writeEndArray()
+		
+	}
+	
+	private void serializeCar(Car car, JsonGenerator jgen) {
+		jgen.writeStartObject()
+		
+		jgen.writeNumberField("carID", car.getPersistenceId())
+		jgen.writeStringField("model", car.getModel())
+		jgen.writeNumberField("buildYear", car.getBuildYear())
+		jgen.writeStringField("color", car.getColor())
+		jgen.writeStringField("wheel1Request", getWheelRequest(car.getPersistenceId(),1))
+		jgen.writeStringField("wheel2Request", getWheelRequest(car.getPersistenceId(),2))
+		jgen.writeStringField("wheel3Request", getWheelRequest(car.getPersistenceId(),3))
+		jgen.writeStringField("wheel4Request", getWheelRequest(car.getPersistenceId(),4))
+		
+		jgen.writeEndObject()
+	}
+	private String getWheelRequest(Long carID, Integer wheelNum) {
+		return String.format('../API/extension/wheel?p=0&c=10&carID=%s&wheelNum=%s', carID, wheelNum)
+	}
+
+}
+
+/***********************
+ ******* MAPPER ********
+ ***********************/
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.databind.module.SimpleModule
+
+class CarObjectMapper extends ObjectMapper {
+	public CarObjectMapper () {
+	    SimpleModule module = new SimpleModule()
+	    module.addSerializer(List.class, new CarSerializer())
+	    registerModule(module)
+    } 
+}
+
+/***********************
+ ******** INDEX ********
+ ***********************/
+
+class CarIndex implements RestApiController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CarIndex.class)
+    private static final CarObjectMapper CAR_MAPPER = new CarObjectMapper()
+
+    @Override
+    RestApiResponse doHandle(HttpServletRequest request, RestApiResponseBuilder responseBuilder, RestAPIContext context) {
+        def p = request.getParameter "p"
+        def c = request.getParameter "c"
+		
+		def carDAO = context.apiClient.getDAO(CarDAO.class)
```

Comment:
indentation
I notice that you have removed all the parameter validation code which was present in previous commits. Is there any reason for that?

Reply:
Yep, it is not the point of this tuto, i think it is better to focus on the subject.

Reply:
ok no pb with that